### PR TITLE
NSD Mitaka - Log meta data

### PR DESCRIPTION
**Pull request by pkharat**

### CODE REVIEW

File: gbpservice/nfp/lib/nfp_log_helper.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                    ('TenantID:' + nf['tenant_id'])
+                    if nf.get('tenant_id')
+                    else ''),
+                '04_service_chain_id': (
+                    ('ServiceChainID:' +
+                     nf['service_chain_id'])
+                    if nf.get('service_chain_id')
+                    else ''),
+                '05_service_chain_instance_id': (
+                    ('ServiceInstanceID:' +
```
Comment (ashutosh-mishra) [inline gbpservice/nfp/lib/nfp_log_helper.py]: ServiceInstanceID is not clear, can we make it as NetworkFunctionID

File: gbpservice/nfp/lib/nfp_log_helper.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        log_info_content = OrderedDict(sorted(log_info_content.items(),
+                                              key=lambda t: t[0]))
+
+        log_msg = '['
+        for content in log_info_content:
+            if log_info_content.get(content):
+                log_msg += log_info_content.get(content) + ', '
+        log_msg = log_msg[:-2]
+        if log_msg:
+            log_msg += '] - '
```
Comment (ashutosh-mishra) [inline gbpservice/nfp/lib/nfp_log_helper.py:80]: Above check is not working, attached logs are
ServiceInstanceID:04966fa1-cef1-4515-8a68-71184905619e]Sending delete NFD config request to configurator

File: gbpservice/nfp/orchestrator/modules/device_orchestrator.py [hunk 3]
```diff
@@ -83,12 +84,16 @@ def __init__(self, conf, controller):
                                   }
 
     def _log_event_created(self, event_id, event_data):
-        LOG.info(_LI("Device Orchestrator, RPC Handler, Created event "
-                     "%s(event_name)s with event data: %(event_data)s"),
-                 {'event_name': event_id, 'event_data': event_data})
+        log_meta_data = (event_data.get('log_meta_data')
+                         if 'log_meta_data' in event_data else '')
```
Comment (ashutosh-mishra) [inline gbpservice/nfp/orchestrator/modules/device_orchestrator.py]: As we are doing get on event_data dict, no need to check for key, we can also do
event_data.get('log_meta_data', '')
Same applicable on all further checks.

File: gbpservice/nfp/lib/nfp_log_helper.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                          event=None, service_type=None,
+                          service_provider=None):
+    """Prepares log meta data string in a specific format
+
+    :param nf - network function object
+    :param log_meta_data - already existing log_meta_data string to restructure
+                           it along with other params like level,event etc
+    :param level - log level
+    :param event_category
+    :param event
```
Comment (ashutosh-mishra) [inline gbpservice/nfp/lib/nfp_log_helper.py]: For params level, event_category, event, Can it be any string ?
For level we can define constants.

File: gbpservice/nfp/lib/nfp_log_helper.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+            }
+
+        log_info_content.update(log_info_nf_content)
+        log_info_content = OrderedDict(sorted(log_info_content.items(),
+                                              key=lambda t: t[0]))
+
+        log_msg = '['
+        for content in log_info_content:
+            if log_info_content.get(content):
+                log_msg += log_info_content.get(content) + ', '
```
Comment (ashutosh-mishra) [inline gbpservice/nfp/lib/nfp_log_helper.py:77]: No need to again do get, inside if check

File: gbpservice/nfp/lib/nfp_log_helper.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        log_msg = log_msg[:-2]
+        if log_msg:
+            log_msg += '] - '
+
+        return log_msg
+    except Exception as err:
+        LOG.error(_LE("Error while generating LOG. %s."
+                      % str(err).capitalize()))
+        raise err
```
Comment (ashutosh-mishra) [inline gbpservice/nfp/lib/nfp_log_helper.py]: This will go as unhandled exception, instead of raising exception  we can return empty string.

**Comment by ashutosh-mishra** [comment]
Looks like attached log file is old, Can you give latest log file
We need to remove this log file, once review is done.

**Comment by njagadish** [comment]
Approach of logging changed and those changes are merged to mainline. This PR is not needed. Closing it.

**Pull request closed without merging**